mongoOperator/helpers/RestoreHelper.py

```diff
@@ -0,0 +1,174 @@
+            backup_file = cluster_object.spec.backups.gcs.restore_from
+            if backup_file == 'latest':
+                backup_file = self.getLastBackup(cluster_object)
+
+            logging.info("Attempting to restore file %s to cluster %s @ ns/%s.", backup_file,
+                         cluster_object.metadata.name, cluster_object.metadata.namespace)
+
+            self.restore(cluster_object, backup_file)
+            return True
+
+        return False
+
+    def restore(self, cluster_object: V1MongoClusterConfiguration, backup_file: str) -> bool:
+        """
+        Attempts to restore the latest backup in the specified location to the given cluster.
+        Creates a new backup for the given cluster saving it in the cloud storage.
+        :param cluster_object: The cluster object from the YAML file.
+        :param backup_file: The filename of the backup we want to restore.
+        """
+        hosts = MongoResources.getConnectionSeeds(cluster_object)
+
+        logging.info("Restoring backup file %s to cluster %s @ ns/%s.", backup_file,
+                     cluster_object.metadata.name, cluster_object.metadata.namespace)
+
+        # Download the backup file from the bucket
+        downloaded_file = self._downloadBackup(cluster_object, backup_file)
+
+        for _ in range(self.RESTORE_RETRIES):
+            # Wait for the replicaset to become ready
+
+            try:
+                logging.info("Running mongorestore --host %s --gzip --archive=%s", ','.join(hosts), downloaded_file)
+                restore_output = check_output(["mongorestore", "--host", ','.join(hosts), "--gzip",
+                                               "--archive=" + downloaded_file])
+                logging.info("Restore output: %s", restore_output)
+                os.remove(downloaded_file)
+                return True
+
+            except CalledProcessError as err:
+                logging.error("Could not restore '{}', attempt {}. Return code: {} stderr: '{}' stdout: '{}'"
+                              .format(backup_file, _, err.returncode, err.stderr, err.stdout))
+                sleep(self.RESTORE_WAIT)
+
+        raise SubprocessError("Could not restore '{}' after {} retries!".format(backup_file, self.RESTORE_RETRIES))
+
+    def _downloadBackup(self, cluster_object: V1MongoClusterConfiguration, backup_file: str) -> str:
+        """
+        Downloads the backup file from cloud storage.
+        :param cluster_object: The cluster object from the YAML file.
+        :param backup_file: The file name of the backup to download.
+        :return: The location of the downloaded file.
+        """
+        prefix = cluster_object.spec.backups.gcs.prefix or self.DEFAULT_BACKUP_PREFIX
+        return self._downloadFile(
+            credentials=self._getCredentials(cluster_object),
+            bucket_name=cluster_object.spec.backups.gcs.restore_bucket \
+                    if cluster_object.spec.backups.gcs.restore_bucket \
```

Comment:
This could be simplified to `bucket_name=cluster_object.spec.backups.gcs.restore_bucket or cluster_object.spec.backups.gcs.bucket`

Reply:
Or, to reduce complexity, make a variable from it first:

```python
bucket = cluster_object.spec.backups.gcs.restore_bucket or cluster_object.spec.backups.gcs.bucket
```